src/CommunityToolkit.Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs

```diff
@@ -430,82 +468,87 @@ private async Task<IReadOnlyDictionary<string, string>> StartOnDemandDaprCompone
 
-            string defaultContent = await File.ReadAllTextAsync(daprDefaultStateStorePath, cancellationToken).ConfigureAwait(false);
-            string newContent = defaultContent.Replace("name: pubsub", $"name: {component.Name}");
+            string newContent = await GetDefaultContent(component, daprAppHostRelativePath, cancellationToken).ConfigureAwait(false);
 
             return await contentWriter(newContent).ConfigureAwait(false);
         }
-        else
-        {
-            _logger.LogInformation("Using in-memory Dapr pub-sub for component '{ComponentName}'.", component.Name);
 
-            return await contentWriter(GetInMemoryPubSubContent(component)).ConfigureAwait(false);
-        }
-    }
+        // If the component is not found in the app host directory, try to get it from the default components directory
+        string daprDefaultStorePath = GetDefaultComponentPath(component.Type);
 
-    private async Task<string> GetStateStoreAsync(DaprComponentResource component, Func<string, Task<string>> contentWriter, CancellationToken cancellationToken)
-    {
-        string userDirectory = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
-        string daprDefaultComponentsDirectory = Path.Combine(userDirectory, ".dapr", "components");
-        string daprDefaultStateStorePath = Path.Combine(daprDefaultComponentsDirectory, "statestore.yaml");
-
-        if (File.Exists(daprDefaultStateStorePath))
+        if (File.Exists(daprDefaultStorePath))
         {
-            _logger.LogInformation("Using default Dapr state store for component '{ComponentName}'.", component.Name);
+            _logger.LogInformation("Using default dapr path for component '{ComponentName}'.", component.Name);
 
-            string defaultContent = await File.ReadAllTextAsync(daprDefaultStateStorePath, cancellationToken).ConfigureAwait(false);
-            string newContent = defaultContent.Replace("name: statestore", $"name: {component.Name}");
+            string newContent = await GetDefaultContent(component, daprDefaultStorePath, cancellationToken).ConfigureAwait(false);
 
             return await contentWriter(newContent).ConfigureAwait(false);
         }
-        else
-        {
-            _logger.LogInformation("Using in-memory Dapr state store for component '{ComponentName}'.", component.Name);
 
-            return await contentWriter(GetInMemoryStateStoreContent(component)).ConfigureAwait(false);
-        }
+        // If the component is not found in the default components directory, use the in-memory secret store
+        _logger.LogInformation("Using in-memory provider for dapr component '{ComponentName}'.", component.Name);
+
+        var content = new DaprComponentSchema(component.Name, inMemoryProvider).ToString();
+        return await contentWriter(content).ConfigureAwait(false);
+    }
+
+    private string GetAppHostRelativePath(string componentName)
```

Comment:
Aside, we're creating a place to store things like this in 9.1, the aspire store.

Reply:
nice -Apphost relative? wanna point me to the code and I could update inline with that?

Reply:
I took a look at the aspire store interface - but it looks like a place to write data to rather than a location for user defined data - the idea of the app relative path is to provide a consistent location for users to place dapr components relative to the project rather than in the userprofile directory